src/clib/cool1d_multi_g.F

```diff
@@ -368,6 +600,44 @@ subroutine cool1d_multi_g(
          enddo
       endif
 
+!     Compute log densities
+
+      logdom = log10(dom)
+      do i = is+1, ie+1
+         if ( itmask(i) ) then
+            logT(i)   = log10(tgas(i))
+            if(icmbTfloor .eq. 1)
+     &         logTcmb(i) = log10(comp2)
+            logrho(i) = log10(d(i,j,k) * dom*mh)
+            if( ispecies .gt. 0) then
+               logH(i)   = log10(HI(i,j,k) * dom)
+               logH2(i)  = log10(HI(i,j,k) * dom)
+            endif
+            if( ispecies .gt. 1 ) then
+              logH2(i)  = log10((HI(i,j,k) + H2I(i,j,k) / 2.d0) * dom)
+              logH2I(i) = log10(H2I(i,j,k) * dom /  2.d0)
+            endif
+            if( ispecies .gt. 2) then
+              logHDI(i) = log10(HDI(i,j,k) * dom /  3.d0)
+            endif
+            if( imcool .eq. 1 ) then
+            if( imchem .eq. 1 ) then
+              logCI(i)  = log10(CI (i,j,k) * dom / 12.d0)
+              logCII(i) = log10(CII(i,j,k) * dom / 12.d0)
+              logOI(i)  = log10(OI (i,j,k) * dom / 16.d0)
+              logCO(i)  = log10(CO (i,j,k) * dom / 28.d0)
+              logOH(i)  = log10(OH (i,j,k) * dom / 17.d0)
+              logH2O(i) = log10(H2O(i,j,k) * dom / 18.d0)
+            endif
+            endif
+            logdvdr(i) = -8.79947961814d0 + 0.5d0 * logrho(i) ! km/s / cm
```

Comment:
This factor comes from Chiaki & Wise (2019), where the velocity gradient is approximated as dv/dr = 1 / (3 * t_ff), where t_ff is the free-fall time.